tests/snappi_tests/multidut/ecn/files/multidut_helper.py

```diff
@@ -181,3 +232,192 @@ def run_ecn_test(api,
+            "flow_name": TEST_FLOW_NAME[0],
+            "flow_dur_sec": DATA_FLOW_DURATION_SEC,
+            "flow_rate_percent": flow_rate_dict,
+            "flow_rate_pps": None,
+            "flow_rate_bps": None,
+            "flow_pkt_size": DATA_FLOW_PKT_SIZE,
+            "flow_pkt_count": None,
+            "flow_delay_sec": DATA_FLOW_DELAY_SEC,
+            "flow_traffic_type": traffic_flow_mode.FIXED_DURATION
+        }
+
+    generate_test_flows(testbed_config=testbed_config,
+                        test_flow_prio_list=test_prio_list,
+                        prio_dscp_map=prio_dscp_map,
+                        snappi_extra_params=snappi_extra_params)
+
+    snappi_extra_params.base_flow_config = base_flow_config2
+
+    snappi_extra_params.traffic_flow_config.data_flow_config = {
+            "flow_name": TEST_FLOW_NAME[1],
+            "flow_dur_sec": DATA_FLOW_DURATION_SEC,
+            "flow_rate_percent": flow_rate_dict,
+            "flow_rate_pps": None,
+            "flow_rate_bps": None,
+            "flow_pkt_size": DATA_FLOW_PKT_SIZE,
+            "flow_pkt_count": None,
+            "flow_delay_sec": DATA_FLOW_DELAY_SEC,
+            "flow_traffic_type": traffic_flow_mode.FIXED_DURATION
+        }
+    generate_test_flows(testbed_config=testbed_config,
+                        test_flow_prio_list=test_prio_list,
+                        prio_dscp_map=prio_dscp_map,
+                        snappi_extra_params=snappi_extra_params)
+
+    flows = testbed_config.flows
+
+    all_flow_names = [flow.name for flow in flows]
+    data_flow_names = [flow.name for flow in flows if PAUSE_FLOW_NAME not in flow.name]
+
+    # Clear PFC and queue counters before traffic run
+    duthost.command("pfcstat -c")
+    duthost.command("sonic-clear queuecounters")
+
+    """ Run traffic """
+    _tgen_flow_stats, _switch_flow_stats, _in_flight_flow_metrics = run_traffic(
+                                                                duthost,
+                                                                api=api,
+                                                                config=testbed_config,
+                                                                data_flow_names=data_flow_names,
+                                                                all_flow_names=all_flow_names,
+                                                                exp_dur_sec=DATA_FLOW_DURATION_SEC +
+                                                                DATA_FLOW_DELAY_SEC,
+                                                                snappi_extra_params=snappi_extra_params)
+
+    post_ctr_3 = get_npu_voq_queue_counters(duthost, dut_port, test_prio_list[0])
+    post_ctr_4 = get_npu_voq_queue_counters(duthost, dut_port, test_prio_list[1])
+
+    ecn_counters = [
+        (init_ctr_3, post_ctr_3),
+        (init_ctr_4, post_ctr_4)
+    ]
+
+    verify_ecn_counters(ecn_counters)
+
+    # Get the current configuration
```

Comment:
better to move this part of the code into a function?

Reply:
done